lib/tasks/generate_pdfs.rake

```diff
@@ -55,6 +55,24 @@ namespace :submission do
 
         PortfolioEvidence.process_new_to_pdf(my_source)
 
+        # Portfolio generation automation:
+        # Flip compile_portfolio to true for all projects where the following conditions are met:
+        # - associated unit is active
+        # - current date is after configured automatic generation dates for the associated unit
+        # - portfolio compilation has not been requested already
+        # - portfolio is still not available
+        # Additionally, set the portfolio_auto_generated property to true to indicate this is auto-generated
+        Project.joins(:unit)
+               .where(units: { active: true })
+               .where("units.portfolio_auto_generation_date < ?", Date.today)
+               .where(compile_portfolio: false)
+               .reject(&:portfolio_available)
+               .each do |project|
```

Comment:
can we use `update_all` here to just run a single update statement on the database?

https://guides.rubyonrails.org/active_record_basics.html#update

Reply:
`portfolio_available` is not a db query, it's a method of the `Project` class. Since this check is mandatory and we don't really want another boolean in the db that can get desynchronized with the filesystem state, we can't use `update_all` here. I am aware this is a performance hit, and I'll be moving this to a new rake task and create a new daily cron job to run it for now to reduce the overhead.